packages/mdc-slider/foundation.ts

```diff
@@ -291,76 +287,68 @@ class MDCSliderFoundation extends MDCFoundation {
     this.handlingThumbTargetEvt_ = false;
     this.setActive_(true);
 
-    const moveHandler = (evt) => {
-      this.handleMove_(evt);
+    const moveHandler = (moveEvent: MouseLikeEvent) => {
+      this.handleMove_(moveEvent);
     };
 
+    const moveEventType = MOVE_EVENT_MAP[downEvent.type as DownEventType];
+
     // Note: upHandler is [de]registered on ALL potential pointer-related release event types, since some browsers
     // do not always fire these consistently in pairs.
     // (See https://github.com/material-components/material-components-web/issues/1192)
     const upHandler = () => {
       this.handleUp_();
-      this.adapter_.deregisterBodyInteractionHandler(MOVE_EVENT_MAP[evt.type], moveHandler);
+      this.adapter_.deregisterBodyInteractionHandler(moveEventType, moveHandler);
       UP_EVENTS.forEach((evtName) => this.adapter_.deregisterBodyInteractionHandler(evtName, upHandler));
     };
 
-    this.adapter_.registerBodyInteractionHandler(MOVE_EVENT_MAP[evt.type], moveHandler);
+    this.adapter_.registerBodyInteractionHandler(moveEventType, moveHandler);
     UP_EVENTS.forEach((evtName) => this.adapter_.registerBodyInteractionHandler(evtName, upHandler));
-    this.setValueFromEvt_(evt);
+    this.setValueFromEvt_(downEvent);
   }
 
   /**
    * Called when the user moves the slider
-   * @param {!Event} evt
-   * @private
    */
-  handleMove_(evt) {
+  private handleMove_(evt: MouseLikeEvent) {
     evt.preventDefault();
     this.setValueFromEvt_(evt);
   }
 
   /**
    * Called when the user's interaction with the slider ends
-   * @private
    */
-  handleUp_() {
+  private handleUp_() {
     this.setActive_(false);
     this.adapter_.notifyChange();
   }
 
   /**
    * Returns the pageX of the event
-   * @param {!Event} evt
-   * @return {number}
-   * @private
    */
-  getPageX_(evt) {
-    if (evt.targetTouches && evt.targetTouches.length > 0) {
-      return evt.targetTouches[0].pageX;
+  private getPageX_(evt: MouseLikeEvent): number {
+    if ((evt as TouchEvent).targetTouches && (evt as TouchEvent).targetTouches.length > 0) {
+      return (evt as TouchEvent).targetTouches[0].pageX;
     }
-    return evt.pageX;
+    return (evt as MouseEvent).pageX;
   }
 
   /**
    * Sets the slider value from an event
-   * @param {!Event} evt
-   * @private
    */
-  setValueFromEvt_(evt) {
+  private setValueFromEvt_(evt: MouseLikeEvent) {
     const pageX = this.getPageX_(evt);
     const value = this.computeValueFromPageX_(pageX);
     this.setValue_(value, true);
   }
 
   /**
    * Computes the new value from the pageX position
-   * @param {number} pageX
-   * @return {number}
    */
-  computeValueFromPageX_(pageX) {
+  private computeValueFromPageX_(pageX: number): number {
     const {max_: max, min_: min} = this;
-    const xPos = pageX - this.rect_.left;
-    let pctComplete = xPos / this.rect_.width;
+    const xPos = pageX - this.rect_!.left; // rect_ is assigned in layout()
```

Comment:
Are the `!`s really needed on `rect_` everywhere even though the member declaration already has it?

Reply:
ah good catch. Those should be removed too.